src/pkg/services/m365/api/graph/service_test.go

```diff
@@ -221,43 +221,92 @@ func (suite *GraphIntgSuite) TestAdapterWrap_catchesPanic() {
 	require.Contains(t, err.Error(), "panic", clues.ToCore(err))
 }
 
-func (suite *GraphIntgSuite) TestAdapterWrap_retriesConnectionClose() {
-	t := suite.T()
-
-	ctx, flush := tester.NewContext(t)
-	defer flush()
-
-	retryInc := 0
-
-	// the panics should get caught and returned as errors
-	alwaysECONNRESET := mwForceResp{
-		err: syscall.ECONNRESET,
-		alternate: func(req *http.Request) (bool, *http.Response, error) {
-			retryInc++
-			return false, nil, nil
+func (suite *GraphIntgSuite) TestAdapterWrap_retriesConnectionInterruptions() {
+	table := []struct {
+		name             string
+		providedErr      error
+		expectRetryCount int
+		expectErr        assert.ErrorAssertionFunc
+	}{
+		{
+			name:             "ECONNRESET",
+			providedErr:      syscall.ECONNRESET,
+			expectRetryCount: 7,
+			expectErr:        assert.Error,
+		},
+		{
+			name:             "connection reset by peer",
+			providedErr:      clues.New("connection reset by peer"),
+			expectRetryCount: 7,
+			expectErr:        assert.Error,
+		},
+		{
+			name:             "read: connection timed out",
```

Comment:
could you please make another test like this but with additional input in the error message? I'm hoping it'll help catch accidental misuse of the filters. For example `clues.New("read: connection timed out: host foo")` or something

Reply:
yes! haha, i was just typing this out. The failed request appends a bunch of extra text to it anyway, but better to have explicit test!

This is how i found out that `filters.In` is what we need instead of `filters.Compare`.